This patch allow Filezilla client to resize window.
The patch was adapted from upstream source repository:
'<http://trac.wxwidgets.org/changeset/4793e5b0a4e189e492287305859b278fed780080/git-wxWidgets>'

--- a/src/gtk/toplevel.cpp	2014-10-06 16:33:44.000000000 -0500
+++ b/src/gtk/toplevel.cpp	2017-02-16 21:33:27.779907810 -0600
@@ -1216,8 +1216,9 @@
     int hints_mask = GDK_HINT_MIN_SIZE | GDK_HINT_MAX_SIZE;
     hints.min_width = 1;
     hints.min_height = 1;
-    hints.max_width = INT_MAX;
-    hints.max_height = INT_MAX;
+    // using INT_MAX for size will lead to integer overflow with HiDPI scaling
+    hints.max_width = INT_MAX / 16;
+    hints.max_height = INT_MAX / 16;    
     const int decorSize_x = m_decorSize.left + m_decorSize.right;
     const int decorSize_y = m_decorSize.top + m_decorSize.bottom;
     if (minSize.x > decorSize_x)
